File: stone_js/web_ui/js/stone_sub.js
// stone_sub.js

'use strict';

// ####################################
// Just an help class
// ####################################

class CallStatistics {
  constructor () {
    this.brick_height = 0;
    this.wall_thermal_conductivity = 0;
  }

  tickOffBrickWeight () {
    this.brick_height += 1;
  }

  tickOffWallThermalConductivity () {
    this.wall_thermal_conductivity += 1;
  }

  getStatistics () {
    // let brick_height_str = this.brick_height.toString();
    // let wall_thermal_conductivity_str = this.wall_thermal_conductivity.toString();
    let total = this.brick_height + this.wall_thermal_conductivity;
    // let total_str = total.toString();
    let r = 'visit counts:\n';
    r += `brick_weight calls                : ${this.brick_height}\n`;
    r += `wall_thermal_conductivity calls   : ${this.wall_thermal_conductivity}\n`;
    r += `total calls                       : ${total}\n`;
    return r;
  }
}

// ####################################
// export ES6-module
// ####################################

export {
  CallStatistics
};

File: stone_py/stone_sub.py
"""
stone_sub.py
Define a class that tracks the function calls of the stone's api
"""

#################
# The helper stuff
#################

class call_statistics:
    """Class to track the activity of the stone functionalities"""

    def __init__(self):
        self.brick_weight = 0
        self.wall_thermal_conductivity = 0

    def tick_off_brick_weight(self):
        self.brick_weight += 1

    def tick_off_wall_thermal_conductivity(self):
        self.wall_thermal_conductivity += 1

    def get_statistics(self):
        r =  "visit counts:\n"
        r += "brick_weight calls                : {}\n".format(self.brick_weight)
        r += "wall_thermal_conductivity calls   : {}\n".format(self.wall_thermal_conductivity)
        r += "total calls                       : {}\n".format(self.brick_weight + self.wall_thermal_conductivity)
        return(r)
